packages/mui-base/src/useSelect/useSelect.ts

```diff
@@ -408,18 +406,44 @@ function useSelect<OptionValue, Multiple extends boolean = false>(
       null) as SelectValue<SelectOption<OptionValue>, Multiple>;
   }
 
+  const createHandleHiddenInputChange =
+    (externalEventHandlers?: EventHandlers) => (event: React.ChangeEvent<HTMLInputElement>) => {
+      externalEventHandlers?.onChange?.(event);
+
+      // support autofill
+      if (options.has(event.target.value as OptionValue)) {
+        const option = options.get(event.target.value as OptionValue);
```

Comment:
I couldn't find a better approach here than the `as OptionValue` assertion. On autofill, the value will be a string; the `if (options.has(...` condition should keep us safe if values are not strings.

Open to suggestions 😅

Reply:
IMO it's OK here. A small optimization we can make is to call options.get and check the return value for `undefined`. This way we avoid looking up the same value twice.

Reply:
Done 👍🏼